packages/client/components/AnalyticsPage.tsx

```diff
@@ -157,7 +144,62 @@ const AnalyticsPage = () => {
       )
     }, TIME_TO_RENDER_TREE)
   }, [isSegmentLoaded, pathname])
+
   return null
 }
 
+function identifyUserWithSegment(viewerInfo?: ViewerInfo) {
+  const segmentKey = window.__ACTION__.segment
+  if (!segmentKey) return
+  if (!viewerInfo) return
+
+  const {id, email} = viewerInfo
+  safeIdentify(id, email)
+}
+
+function setupLogRocket(viewerInfo?: ViewerInfo) {
```

Comment:
Currently, there are four distinct use cases where we init, identify and report to LogRocket. In all of these examples, "error prone" means "encountered an error that bubbled to the Error Boundary within the last 14 days".

1. Logged out and error prone
2. Logged in and error prone
3. Logged in and watched
4. Any user who encounters an error that bubbles the ErrorBoundary (this is also where the Error Prone flag is set).

This logic could be simplified if we simply initialized for all users, identifying the user based on the `viewerInfo` properties set. However, I'm assuming the reason we don't do this today is because LogRocket charges per-session. 

To make this all simpler, I created a simple `LogRocketManager` class. This allows me to call `setUser` on the class whenever it changes. Then, whenever the LogRocket SDK is initialized, it'll use whatever user is current upon initialization.

Reply:
That's correct! LogRocket charges per session so we created the errorProne property to reduce costs. Nice work simplifying this